src/Features/Core/Navigation/IDocumentNavigationService.cs

```diff
@@ -25,16 +25,16 @@ internal interface IDocumentNavigationService : IWorkspaceService
         /// <summary>
         /// Navigates to the given position in the specified document, opening it if necessary.
         /// </summary>
-        bool TryNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan, bool usePreviewTab = false);
+        bool TryNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan, bool reopenDocument = false, bool usePreviewTab = false);
```

Comment:
We don't have context about whether we should be opening in a preview or permanent tab when NavigateTo calls us, we only have the agreement that they have set the underlying new document state appropriately. Just because a NavigateTo item's document is already open in a preview tab and then requested to be navigated to again doesn't mean it's ready to be transitioned to a permanent tab (the user could just be previewing the same item again or something).